### PR TITLE
feat: apply local UI library

### PR DESCRIPTION
**Pull request by VanishMax**
This PR inserts the UI package (`src` directory) into the DEX to unblock faster local development. Adds Storybook config to the DEX

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
       }
     ],
     "paths": {
-      "@/*": ["./src/*", "./styles/*"]
+      "@/*": ["./src/*"],
+      "@penumbra-zone/ui/*": ["./src/shared/ui/*"]
```
Comment (VanishMax) [inline tsconfig.json:16]: this allows importing `@penumbra-zone/ui/Component` from anywhere in the DEX code

File: package.json [hunk 2]
```diff
@@ -21,89 +21,89 @@
...
     "react-dom": "^18.3.1",
-    "react-outside-click-handler": "^1.3.0",
-    "sharp": "^0.33.5",
-    "webpack": "^5.94.0",
+    "tinycolor2": "^1.6.0",
+    "webpack": "^5.95.0",
     "zod": "^3.23.8"
   },
   "devDependencies": {
-    "@babel/core": "^7.25.2",
```
Comment (VanishMax) [inline package.json:79]: - removed legacy dependencies
- updated existing dependencies
- added many libraries needed to support the UI package

File: package.json [hunk 3]
```diff
@@ -17,93 +17,104 @@
...
+    "@storybook/addon-essentials": "^8.1.1",
+    "@storybook/addon-interactions": "^8.1.1",
+    "@storybook/addon-links": "^8.1.1",
+    "@storybook/addon-postcss": "^2.0.0",
+    "@storybook/addon-webpack5-compiler-swc": "^1.0.5",
+    "@storybook/blocks": "^8.1.1",
+    "@storybook/manager-api": "^8.1.11",
+    "@storybook/preview-api": "^8.1.1",
+    "@storybook/react": "^8.1.1",
+    "@storybook/react-webpack5": "^8.3.6",
```
Comment (VanishMax) [inline package.json:88]: different Storybook setup comparing to the Web repo – relies on webpack instead of Vite

**Review by TalDerei — CHANGES_REQUESTED** [review]
proposing closing in favor of https://github.com/penumbra-zone/dex-explorer/issues/115

**Comment by VanishMax** [comment]
@TalDerei https://github.com/penumbra-zone/dex-explorer/pull/116

**Pull request closed without merging**